modal/object.py

```diff
@@ -204,9 +204,12 @@ async def resolve(self):
                 " wasn't defined in global scope."
             )
         else:
-            resolver = Resolver()
+            resolver = Resolver()  # TODO: this resolver has no attached Client!
```

Comment:
Added this TODO, as I noticed this is the only place we initialize a Resolver without a working client, which caused issues for lazily hydrated objects until I patched my way around it. Would still cause issues in the future if a lazy object used the resolver to load additional entities that relies on the client